fh-sync-js.d.ts

```diff
@@ -426,12 +426,20 @@ declare module SyncClient {
    * @param handler - function that wraps underlying storage solution
    */
   function setStorageAdapter(handler: (dataset_id: string, isSave: boolean, cb: any) => void): void;
+
   /**
    * Allows to override default hashing method
    *
    * @param method - function that wraps underlying hashing method
    */
   function setHashMethod(method: () => any): void;
+
+  /**
+   * Allows to override default method for checking if application is online
+   *
+   * @param method - function that checks if network is available
+   */
+  function setNetworkStatusHandler(method: (callback: (isOnline: boolean) => void) => void): void;
```

Comment:
I agree that we shouldn't mix promises in occasionally. The API should support one or the other, or both, but not a half-hearted mix. And for the record, promisify won't work with most sync (and other FH) functions, as they don't follow the standard single function `(err, result)` format.

That being said, my professional opinion is that you guys are going to be on the wrong side of history in callbacks vs promises. Most (popular, larger) libraries I see, if they support callbacks, generally support promises as well: if you provide a callback, it'll use that, otherwise it'll return a Promise. `mongodb` does this. It's really trivial if the inner workings of your library are promise-based, but mongodb also accomplishes it even though most of their internals are callback-based.

In my experience with people I've talked to, callbacks are generally deemed "legacy code" these days. There's a reason the `mz` library has nearly a million downloads per month. `request-promise` has double that. In addition to all the benefits of promises that Evan and I have probably stated ad nauseam, native async/await is just around the corner. Many shops have already been using them successfully with either Babel or TS, and callbacks-only API is completely incompatible with them.